target/scripts/check-for-changes.sh

```diff
@@ -6,37 +6,34 @@
 # shellcheck source=./helpers/index.sh
 source /usr/local/bin/helpers/index.sh
 
-function _log_date
-{
-  date +"%Y-%m-%d %H:%M:%S"
-}
-
 function _changedetector_notify
 {
-  local LOG_LEVEL_ARGUMENT="${1}"
+  local LEVEL="${1}"
   shift 1
 
-  _notify "${LOG_LEVEL_ARGUMENT}" '[ CHANGEDETECTOR ]' " $(_log_date) " "${*}"
+  _notify "${LEVEL}" '[ CHANGEDETECTOR ]' " $(date +"%Y-%m-%d %H:%M:%S") " "${*}"
 }
 
-LOG_DATE=$(_log_date)
-_changedetector_notify 'debug' "${LOG_DATE} Start check-for-changes script."
+_changedetector_notify 'debug' 'Starting changedetector'
 
 # ? --------------------------------------------- Checks
 
-cd /tmp/docker-mailserver || exit 1
+if ! cd /tmp/docker-mailserver &>/dev/null
+then
+  _changedetector_notify 'error' "Could not change into '/tmp/docker-mailserver' directory"
+  exit 1
+fi
```

Comment:
Just so I understand, we're still changing the directory here, but that command is part of a negative conditional now?

Would you say the failure here warrants the `exit 1` still while the other two checks below use `exit 0`? (_and explicitly state that the script will exit in the log_) Should they all be consistent or is there a difference? The other two also use an `info` level instead of `error` which is perhaps wrong but why they had `exit 0`?

Reply:
> Just so I understand, we're still changing the directory here, but that command is part of a negative conditional now?

Exactly, we're still changing the directory. I added this to make it more clear why the script exited with exit code 1.
 
> Would you say the failure here warrants the `exit 1` still while the other two checks below use `exit 0`? (_and explicitly state that the script will exit in the log_) Should they all be consistent or is there a difference? The other two also use an `info` level instead of `error` which is perhaps wrong but why they had `exit 0`?

I think so, because this really seems like an unrecoverable error to me. A `notify 'error'` should always be followed by an `exit` or `return` call with non-zero exit code (to properly follow the semantics of the `error` log level).

~~I did not touch the other two statements because I really am not sure whether they are errors or just, let's say LDAP related with `postfix-accounts.cf`. I think both could indeed be errors, and this would mean adjusting the log level to `error` and `exit`ing with `1`.~~

Okay, so according to

https://github.com/docker-mailserver/docker-mailserver/blob/b730942b96aec3ab52699b5d927577f42eef0a45/target/scripts/start-mailserver.sh#L197

this cannot be LDAP related. Consequently, these have to be errors.

Reply:
> I think both could indeed be errors, and this would mean adjusting the log level to error and exiting with 1.

Both result in the change detector exiting early instead of running like intended. I'd consider that an error?

> this cannot be LDAP related. Consequently, these have to be errors.

Yeah, but at some point we do want to support with LDAP enabled, at least for the cert renewal support :sweat_smile:

Reply:
Done in cce33f80d545c23eaddb67a4afb2a2bd9596e0d9